### ISSUE TITLE
Social Meta - OG Image not pulling in full size image

### ISSUE THREAD

**Issue by kevinc08** [awesomemotive/all-in-one-seo-pack#260]
The OG image section in AIOSEO no longer pulls in the full size image but rather the large 1024x538 image (if the original image is bigger than that).

Previously, AIOSEO would just show the original full size image as one of the options in the image section.

https://scontent-hkg3-1.xx.fbcdn.net/hphotos-xtp1/v/t34.0-12/13152821_10153477949676837_522676256_n.png?oh=cbdbee3a560316cb4f817db447d614a3&oe=572CFEBA

I get around it by uploading a custom image and setting it to full size to use as my facebook OG image however this is tedious when I have to do a lot of posts. 

Could you please change it back so AIOSEO pulls in the full size image please?

**Comment by michaeltorbert** [comment]
![13152821_10153477949676837_522676256_n](https://cloud.githubusercontent.com/assets/17364998/15024234/13c749a8-1202-11e6-94cb-561b06f71032.png)

**Comment by arnaudbroes** [comment]
Looks like it's fixed. :-)

**Comment by michaeltorbert** [comment]
@arnaudbroes I don't think this has been addressed yet.

**Comment by wpsmort** [comment]
I have tested this and it only happens with images when you select either the Featured Image or the Attached Image options under All in One SEO > Social Meta > Image Settings > Select OG:Image Source (https://semperplugins.com/documentation/social-meta-module/#select-og-image-source).

**Comment by wpsmort** [comment]
Reported again here - https://wordpress.org/support/topic/open-graph-featured-image-is-wrong-size/

**Comment by Garconis** [comment]
For now, I added:

```
add_filter( 'aioseop_thumbnail_size', 'change_aioseop_thumbnail_size' , 10, 1);
function change_aioseop_thumbnail_size( $size ){
    return 'full';
}
```

As documented here: https://semperplugins.com/documentation/opengraph-image-size/

But this should be fixed to at least offer an option to set the size, since having it use the "Large" size isn't ideal, nor expected.

**Comment by Garconis** [comment]
It seems the Featured Image is no longer being pulled in at all?

Settings: https://i.gyazo.com/a7b7e46f3db18da59542b721a73ae021.png

Yet, new posts that do have featured images, the opengraph tag is wrong. :(

This seems to have started less than a month ago.

**Comment by contactashish13** [comment]
PR: https://github.com/semperfiwebdesign/all-in-one-seo-pack/pull/1461

While testing, I've found what could potentially be a bug: https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/1462

**Comment by wpsmort** [comment]
@contactashish13 I tested PR #1461 and found two problems:

1.  This new code overrides the filters documented here - https://semperplugins.com/documentation/opengraph-image-size/.  Filters applied through the API should override the plugin.  

2.  This bug is present if you select First Available Image under Social Meta > Image Settings > Select OG Image Source.  When this is selected it uses the Featured Image large size.  If there's no Featured Image then it uses the first Attached Image large size.

**Comment by contactashish13** [comment]
@wpsmort have addressed this.

**Comment by wpsmort** [comment]
@contactashish13 I tested the latest code in the PR and still have the following problems:

1. When you select First Available Image under Social Meta > Image Settings > Select OG Image Source the image size used is the large image.

2. When you select either First Attached Image of First Image in Content then the aioseop_thumbnail_size filter is ignored.

**Comment by contactashish13** [comment]
@wpsmort I've tested all combinations with the filter.

1. have fixed this
2a. For First Attached Image, the filter being used was `aioseop_attachment_size`. Now for all cases I'm using 3 filters in the following way
`
$size  = apply_filters( 'aioseop_attachment_size', apply_filters( 'aioseop_thumbnail_size', apply_filters( 'post_thumbnail_size', 'large' ) ) );
`
2b. Have fixed this.

**Comment by wpsmort** [comment]
Thanks @contactashish13. Can you look at the Codeclimate and Travis checks that failed please.

**Comment by wpsmort** [comment]
@contactashish13 I tested this and it still overriders the filter hook if you select either First Attached Image or First Available Image.  The filter works fine if you select Featured Image.

**Comment by contactashish13** [comment]
@wpsmort have tested all combinations and they all work.

**Comment by michaeltorbert** [comment]
In 2.9 we introduced a fix for this, but then reverted in 2.9.1 due to performance issues.

Original PR https://github.com/semperfiwebdesign/all-in-one-seo-pack/pull/1461
Revert PR https://github.com/semperfiwebdesign/all-in-one-seo-pack/pull/2007
Revert Issue https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/1997

**Comment by wpsmort** [comment]
@EkoJR I'm assigning this to you so you can integrate this into v4.

**Comment by wpsmort** [comment]
Reported again here - https://wordpress.org/support/topic/override-aiosep_attachment_size/

**Comment by arnaudbroes** [comment]
Fixed a while ago in https://github.com/awesomemotive/aioseo/issues/955

**Comment by arnaudbroes** [comment]
> > Fixed a while ago in [awesomemotive/aioseo#955](https://github.com/awesomemotive/aioseo/issues/955)
> 
> Page not found. But I assume you're referring to the update of 4.0.13 via:
> 
> * Fixed: Use fullsize image for OG image with filter to modify `aioseo_thumbnail_size`

Correct. Like I commented on the other issue, this was just for our own internal reference.
